app/Services/Model/Imp/SummitService.php

```diff
@@ -1817,50 +1817,134 @@ public function cloneEvent(Summit $summit, int $event_id): SummitEvent
-                $eventClone->setRSVPLink($event->getRSVPLink());
+            $external_id = $event->getExternalId();
+            if ($external_id != null) {
+                $event_clone->setExternalId($external_id);
             }
 
+            $location = $event->getLocation();
+            if ($location != null) {
+                $event_clone->setLocation($location);
+            }
+
+            $local_start_date = $event->getLocalStartDate();
+            if ($local_start_date != null) {
+                $event_clone->setStartDate($local_start_date);
+            }
+
+            $local_end_date = $event->getLocalEndDate();
+            if ($local_end_date != null) {
+                $event_clone->setEndDate($local_end_date);
+            }
+
+            $category = $event->getCategory();
+            if ($category != null) {
+                $event_clone->setCategory($category);
+            }
+
+            $event_clone->setStreamingUrl($event->getStreamingUrl());
+
+            $etherpad_link = $event->getEtherpadLink();
+            if ($etherpad_link != null) {
+                $event_clone->setEtherpadLink($etherpad_link);
+            }
+
+            $meeting_url = $event->getMeetingUrl();
+            if ($meeting_url != null) {
+                $event_clone->setMeetingUrl($meeting_url);
+            }
+
+            if ($event->hasImage()) {
+                $event_clone->setImage($event->getImage());
+            }
+
+            $mux_playback_id = $event->getMuxPlaybackId();
+            if ($mux_playback_id != null) {
+                $event_clone->setMuxPlaybackId($mux_playback_id);
+            }
+
+            $mux_asset_id = $event->getMuxAssetId();
+            if ($mux_asset_id != null) {
+                $event_clone->setMuxAssetId($mux_asset_id);
+            }
+
+            $level = $event->getLevel();
+            if ($level != null) {
+                $event_clone->setLevel($level);
+            }
+
+            $current_user = ResourceServerContext::getCurrentUser(false);
+
+            $event_clone->setCreatedBy($current_user);
+            $event_clone->setUpdatedBy($current_user);
+            $event_clone->setStreamIsSecure($event->IsSecureStream());
+
             foreach ($event->getSponsors() as $sponsor) {
-                $eventClone->addSponsor($sponsor);
+                $event_clone->addSponsor($sponsor);
             }
 
             foreach ($event->getTags() as $tag) {
-                $eventClone->addTag($tag);
+                $event_clone->addTag($tag);
             }
 
             // check if SummitEventWithFile
 
             if ($event instanceof SummitEventWithFile && $event->hasAttachment()) {
-                $eventClone->setAttachment($event->getAttachment());
+                $event_clone->setAttachment($event->getAttachment());
             }
 
-            $this->event_repository->add($eventClone);
+            if ($event instanceof Presentation) {
```

Comment:
also need special cases fro following derived classes
* SummitGroupEvent
* SummitEventWithFile

Reply:
SummitEventWithFile already had special treatment (https://github.com/OpenStackweb/summit-api/pull/199/files#diff-a4e3ced07cbae056279455d51c16cb0ce5d06fa73af7c022b5c40fac42af10bfR1907)

I added the SummitGroupEvent part